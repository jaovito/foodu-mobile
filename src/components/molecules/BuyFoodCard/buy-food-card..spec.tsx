import {fireEvent, render} from '@test-utils';
import {mock} from '../../../test/mocks';
import {BuyFoodCard} from '.';

describe('BuyFoodCard tests', () => {
  it('should be defined', () => {
    const card = render(
      <BuyFoodCard
        name={mock.buyFoodCard.name}
        about={mock.buyFoodCard.about}
        price={mock.buyFoodCard.price}
      />,
    );

    expect(card).toBeDefined();
  });

  it('should be able to find the name of the card', async () => {
    const {findByText} = render(
      <BuyFoodCard
        name={mock.buyFoodCard.name}
        about={mock.buyFoodCard.about}
        price={mock.buyFoodCard.price}
      />,
    );

    const cardName = await findByText(mock.buyFoodCard.name);

    expect(cardName).toBeTruthy();
  });

  it('should be able to find the about text of the card', async () => {
    const {findByText} = render(
      <BuyFoodCard
        name={mock.buyFoodCard.name}
        about={mock.buyFoodCard.about}
        price={mock.buyFoodCard.price}
      />,
    );

    const cardAboutText = await findByText(mock.buyFoodCard.about);

    expect(cardAboutText).toBeTruthy();
  });

  it('should be able to click on the card', async () => {
    const onPress = jest.fn();

    const {getByText} = render(
      <BuyFoodCard
        name={mock.buyFoodCard.name}
        about={mock.buyFoodCard.about}
        price={mock.buyFoodCard.price}
        onPress={onPress}
      />,
    );

    fireEvent.press(getByText(mock.buyFoodCard.name));

    expect(onPress).toBeCalled();
  });

  it('should match with snapshot', async () => {
    const onPress = jest.fn();

    const card = render(
      <BuyFoodCard
        name={mock.buyFoodCard.name}
        about={mock.buyFoodCard.about}
        price={mock.buyFoodCard.price}
        onPress={onPress}
      />,
    ).toJSON();

    expect(card).toMatchSnapshot();
  });
});
