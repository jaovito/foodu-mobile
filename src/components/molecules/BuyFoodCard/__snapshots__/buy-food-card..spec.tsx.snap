// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyFoodCard tests should match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "rgba(0, 0, 0, 0.1)",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 2,
      "justifyContent": "center",
      "opacity": 1,
      "width": 304,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../src/assets/defaultChef.png",
      }
    }
    style={
      [
        {
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "height": 196,
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      [
        {},
      ]
    }
  >
    <Text
      fontWeight="semiBold"
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_600SemiBold",
            "fontSize": 31,
          },
          {
            "fontSize": 29,
            "lineHeight": 35,
            "marginTop": 16,
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 0,
          },
        ]
      }
    >
      Chicken burger
    </Text>
    <Text
      numberOfLines={2}
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 31,
          },
          {
            "fontSize": 22,
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 0,
          },
        ]
      }
    >
      200 gr chicken + cheese  Lettuce + tomato
    </Text>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 16,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        fontWeight="bold"
        style={
          [
            {
              "color": "#0D0D0D",
              "fontFamily": "Poppins_700Bold",
              "fontSize": 31,
            },
            {
              "color": "#FE8C00",
              "fontSize": 27,
            },
          ]
        }
      >
        $
        20.00
      </Text>
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FE8C00",
              "borderRadius": 24,
              "height": 47,
              "justifyContent": "center",
              "width": 47,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              [
                {
                  "color": "#FFFFFF",
                  "fontSize": 27,
                },
              ],
              {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          ï‡€
        </Text>
      </View>
    </View>
  </View>
</View>
`;
